mailpoet/assets/js/src/segments/dynamic/list/bulk-actions.tsx

```diff
@@ -0,0 +1,202 @@
+import { useState } from 'react';
+import { __, _n, sprintf } from '@wordpress/i18n';
+import { SelectControl } from '@woocommerce/components';
+import { Button, Modal } from '@wordpress/components';
+import { dispatch, select } from '@wordpress/data';
+import { MailPoet } from '../../../mailpoet';
+import { DynamicSegment } from '../types';
+import { storeName } from '../store';
+
+type ConfirmModalProps = {
+  onClose: () => void;
+  onConfirm: () => void;
+  confirmText: string;
+  title: string;
+  message: string | JSX.Element;
+};
+function ConfirmModal({
+  onClose,
+  onConfirm,
+  confirmText,
+  title,
+  message,
+}: ConfirmModalProps): JSX.Element {
+  return (
+    <Modal title={title} onRequestClose={onClose}>
+      <p>{message}</p>
+      <Button variant="primary" onClick={onConfirm}>
+        {confirmText}
+      </Button>
+      <Button variant="tertiary" onClick={onClose}>
+        {__('Cancel', 'mailpoet')}
+      </Button>
+    </Modal>
+  );
+}
+
+async function bulkAction(action: string, segments: DynamicSegment[]) {
+  void MailPoet.Ajax.post({
+    api_version: 'v1',
+    endpoint: 'dynamic_segments',
+    action: 'bulk_action',
+    data: {
+      action,
+      listing: {
+        selection: segments.map((segment) => segment.id),
+      },
+    },
+  }).then(() => {
+    void dispatch(storeName).loadDynamicSegments();
+  });
+}
+
+type BulkActionsProps = {
+  tab: {
+    name: string;
+  };
+};
+export function BulkActions({ tab }: BulkActionsProps): JSX.Element {
+  const [currentAction, setCurrentAction] = useState<string>('');
+  const dynamicSegments = select(storeName).getDynamicSegments();
+  const allSelected = dynamicSegments
+    ? dynamicSegments.filter((segment) => segment.selected)
+    : [];
+  const bulkActions =
+    tab.name !== 'trash'
+      ? [
+          {
+            label: __('Trash', 'mailpoet'),
+            value: 'trash',
+          },
+        ]
+      : [
+          {
+            label: __('Restore', 'mailpoet'),
+            value: 'restore',
+          },
+          {
+            label: __('Delete permanently', 'mailpoet'),
+            value: 'delete',
+          },
+        ];
+
+  let title = '';
+  let message: string | JSX.Element = '';
+  let confirmText = '';
+  const lastSelected =
+    allSelected.length > 0
+      ? `"${allSelected[allSelected.length - 1].name}"`
+      : '';
+  const firstSelected =
+    allSelected.length > 1
+      ? allSelected
+          .slice(0, -1)
+          .map((segment) => `"${segment.name}"`)
+          .join(', ')
+      : lastSelected;
```

Comment:
Generally, for these things there is the [Intl.ListFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/ListFormat), but I guess no need to necessarily change now.

```ts
const list = new Intl.ListFormat(language, { style: 'long', type: 'conjunction'})
  .format(allSelected.map((value) => `"${value}"`));
```